3pp/compat/mbedtls/library/mbedtls_x509write_crt.c

```diff
@@ -320,32 +324,32 @@ int mbedtls_x509write_crt_der( mbedtls_x509write_cert *ctx, unsigned char *buf,
     if( ( ret = mbedtls_oid_get_oid_by_sig_alg( pk_alg, ctx->md_alg,
                                         &sig_oid, &sig_oid_len ) ) != 0 )
     {
-        return( ret );
+        return( val );
     }
 
     /*
      *  Extensions  ::=  SEQUENCE SIZE (1..MAX) OF Extension
      */
-    MBEDTLS_ASN1_CHK_ADD( len, mbedtls_x509_write_extensions( &c, tmp_buf, ctx->extensions ) );
-    MBEDTLS_ASN1_CHK_ADD( len, mbedtls_asn1_write_len( &c, tmp_buf, len ) );
-    MBEDTLS_ASN1_CHK_ADD( len, mbedtls_asn1_write_tag( &c, tmp_buf, MBEDTLS_ASN1_CONSTRUCTED |
+    MBEDTLS_ASN1_CHK_ADD2( len, mbedtls_x509_write_extensions( &c, tmp_buf, ctx->extensions ) );
```

Comment:
instead of returning "-1" regardless of the error encountered, i'm returning an integer which represesnts "CHK_ADD2" line where it crashed.
used me a lot during debugging, i consider it stable now, should i revert to ADD instead of ADD2?